src/commissaire_service/storage/__init__.py

```diff
@@ -108,8 +126,82 @@ def register_store_handler(self, config):  # pragma: no cover (temporary)
                     'No match for model: {}'.format(pattern))
             matched_types.update([self._model_types[name] for name in matches])
 
-        self._manager.register_store_handler(
-            handler_type, config, *matched_types)
+        handler_type.check_config(config)
+
+        definition = (handler_type, config, matched_types)
+
+        name = config.get('name', '').strip()
+        if not name:
+            # If the store handler definition was not given a
+            # name, derive a unique name from its module name.
+            suffix = 1
+            name = handler_type.__module__
+            while name in self._definitions_by_name:
+                name = '{}-{}'.format(handler_type.__module__, suffix)
+                suffix += 1
+            config['name'] = name
+
+        if name in self._definitions_by_name:
+            raise ConfigurationError(
+                'Duplicate storage handlers named "{}"'.format(name))
+
+        for mt in matched_types:
+            if mt in self._definitions_by_model_type:
+                conflicting_type, _, _ = \
+                    self._definitions_by_model_type[mt]
+                raise ConfigurationError(
+                    'Model "{}" already assigned to "{}"'.format(
+                        getattr(mt, '__name__', '?'),
+                        getattr(conflicting_type, '__module__', '?')))
+
+        # Add definition after all checks pass.
+        self._definitions_by_name[name] = definition
+        new_items = {mt: definition for mt in matched_types}
+        self._definitions_by_model_type.update(new_items)
+
+    def _create_handler(self, definition):
+        """
+        Creates a handler instance from a handler definition, and adds the
+        handler instance to various internal data structures.
+        """
+        handler_type, config, model_types = definition
+        handler = handler_type(config)
+        handler.notify.connect(self._exchange, self._channel)
```

Comment:
I can't find a ``notify`` attribute that binds [connect](https://github.com/projectatomic/commissaire/pull/121/files#diff-a522d70a10478cbbd1abc5be45f1f7e1R38) to the handler with the exception of [ConfigurationError](https://github.com/projectatomic/commissaire/pull/121/files#diff-a522d70a10478cbbd1abc5be45f1f7e1R38). How does it get wired?